aptos-move/framework/aptos-stdlib/sources/cryptography/multi_ed25519.move

```diff
@@ -191,47 +225,72 @@ module aptos_std::multi_ed25519 {
-        ];
-
-        let num = vector::length(&ns);
-        let i = 0;
-        while (i < num) {
-            let pk_bytes = *vector::borrow(&pks, i);
-            let sig_bytes = *vector::borrow(&sigs, i);
-
-            let pk = new_validated_public_key_from_bytes(pk_bytes);
-            let pk = std::option::extract(&mut pk);
-            let pk = public_key_into_unvalidated(pk);
-            let sig = new_signature_from_bytes(sig_bytes);
-
-            assert!(signature_verify_strict(&sig, &pk, msg), 1);
-
-            i = i + 1;
+    fun test_gen_sign_verify() {
+        let thresholds = vector[1, 1, 2, 2, 3, 15,]; // the thresholds, implicitly encoded in the public keys
+        let party_counts = vector[1, 2, 2, 3, 10, 32,];
+        let test_case_count = vector::length(&party_counts);
+        let test_case_idx = 0;
+        while (test_case_idx < test_case_count) {
+            let threshold = *vector::borrow(&thresholds, test_case_idx);
+            let group_size = *vector::borrow(&party_counts, test_case_idx);
+            let (sk, pk) = generate_keys(threshold, group_size);
+            let upk = public_key_into_unvalidated(pk);
+            let msg1 = b"Hello Aptos!";
+            let sig1 = multi_sign_arbitrary_bytes(&sk, msg1);
+            assert!(signature_verify_strict(&sig1, &upk, msg1), error::invalid_state(1));
+
+            let obj2 = TestMessage {
+                foo: b"Hello Move!",
+                bar: 64,
+            };
+            let sig2 = multi_sign_struct(&sk, copy obj2);
+            assert!(signature_verify_strict_t(&sig2, &upk, copy obj2), error::invalid_state(2));
+
+            test_case_idx = test_case_idx + 1;
         }
     }
+
+    #[test]
+    fun test_threshold_not_met_rejection() {
+        let (sk,pk) = generate_keys(4, 5);
+        let upk = public_key_into_unvalidated(pk);
+
+        let msg1 = b"Hello Aptos!";
+        let sig1 = multi_sign_arbitrary_bytes(&sk, msg1);
+        maul_first_signature(&mut sig1);
+        assert!(!signature_verify_strict(&sig1, &upk, msg1), error::invalid_state(3));
```

Comment:
In tests, we usually don't wrap error codes in `std::error`. I tend to just use 1 all over the place, but using different numbers is a good idea too, to help you tell which `assert` failed. I suppose wrapping error codes in `std::error` might even make this more difficult.